cli/src/etos_client/logs/logs.py

```diff
@@ -157,7 +158,9 @@ def stop(self, clear_queue=True):
 
     def __queue_download(self, item: Downloadable) -> None:
         """Queue a downloadable for download."""
-        self.__download_queue.put_nowait(item)
+        if item.uri not in self.__downloaded:
+            self.__download_queue.put_nowait(item)
+            self.__downloaded.append(item.uri)
```

Comment:
The `LogDownloader` will attempt to download all logs in the queue, if a log fails to download it will retry for 30 seconds.
If `stop` is called, then the `LogDownloader` will wait for the queue to empty before shutting down, unless it has been forced to shut down without clearing it.
The `__downloaded` list is to track files that don't have to be added to the queue again and the `__queue_download` method is not called from threads so it will never have a race condition with itself.

I will rename the `__downloaded` list to make it clear that the files are not downloaded, but queued.